ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -337,12 +341,11 @@ public void onMessage(final Packet packet, final DiscoveryPeer sender) {
         break;
       case ENR_REQUEST:
         if (PeerDiscoveryStatus.BONDED.equals(peer.getStatus())) {
-          LOG.trace("ENR_REQUEST received from bonded peer Id: {}", peer.getId());
-          packet
-              .getPacketData(ENRRequestPacketData.class)
-              .ifPresent(p -> respondToENRRequest(p, packet.getHash(), peer));
+          processEnrRequest(peer, packet);
+        } else if (PeerDiscoveryStatus.BONDING.equals(peer.getStatus())) {
```

Comment:
Also probably worth adding some tests around this.

Reply:
added a test for the cache